src/common/histogram.c

```diff
@@ -29,332 +29,163 @@
 #include "common/histogram.h"
 #include "develop/imageop.h"
 
-#define S(V, params) ((params->mul) * ((float)V))
-#define P(V, params) (CLAMP((V), 0, (params->bins_count - 1)))
-#define PU(V, params) (MIN((V), (params->bins_count - 1)))
-#define PS(V, params) (P(S(V, params), params))
+typedef void((*_histogram_worker)(const dt_dev_histogram_collection_params_t *const params,
+                                  const void *const restrict pixel,
+                                  uint32_t *const restrict histogram, const int j,
+                                  const dt_iop_order_iccprofile_info_t *const profile));
 
-//------------------------------------------------------------------------------
-
-inline static void histogram_helper_cs_RAW_helper_process_pixel_float(
-    const dt_dev_histogram_collection_params_t *const histogram_params, const float *pixel, uint32_t *histogram)
+static inline void _clamp_bin(const dt_aligned_pixel_t vals,
+                              uint32_t *const restrict histogram,
+                              const float max_bin)
 {
-  const uint32_t i = PS(*pixel, histogram_params);
-  histogram[4 * i]++;
-}
-
-inline static void histogram_helper_cs_RAW(const dt_dev_histogram_collection_params_t *const histogram_params,
-                                           const void *pixel, uint32_t *histogram, int j,
-                                           const dt_iop_order_iccprofile_info_t *const profile_info)
-{
-  const dt_histogram_roi_t *roi = histogram_params->roi;
-  const float *input = (float *)pixel + roi->width * j + roi->crop_x;
-  for(int i = 0; i < roi->width - roi->crop_width - roi->crop_x; i++, input++)
-  {
-    histogram_helper_cs_RAW_helper_process_pixel_float(histogram_params, input, histogram);
-  }
+  DT_ALIGNED_PIXEL size_t bin[4];
+  for_each_channel(k,aligned(vals,bin:16))
```

Comment:
please enclose with `{ }` the loop instructions. With the comment in between the first line of code it is a source of errors for maintenance.